### PR TITLE
fix build for all architectures

### PR DESCRIPTION
**Pull request by Stebalien**
This also adds a handy script to test building on all supported architecture/tag combinations.

Note: I'm not sure about my second commit but the build fails without that.

fixes https://github.com/multiformats/go-multihash/issues/110

### CODE REVIEW

File: test-architectures.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+set -e
+
+go tool dist list | while IFS=/ read os arch; do
+    echo "Checking $os/$arch..."
+    echo " normal"
+    GOARCH=$arch GOOS=$os go build -o /dev/null ./...
+    echo " noasm"
+    GOARCH=$arch GOOS=$os go build -tags noasm -o /dev/null ./...
+    echo " appengine"
+    GOARCH=$arch GOOS=$os go build -tags noasm -o /dev/null ./...
```
Comment (Kubuxu) [inline test-architectures.sh]:
```suggestion
    GOARCH=$arch GOOS=$os go build -tags appengine -o /dev/null ./...
```

File: sha256blockAvx2_amd64.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//+build !noasm
```
Comment (Stebalien) [inline sha256blockAvx2_amd64.go]: I assume these need to be disabled if either appengine or asm are passed.

File: sha256block_arm64.s [hunk 3]
```diff
@@ -1,4 +1,4 @@
-//+build !noasm !appengine
+//+build !noasm,!appengine
```
Comment (Stebalien) [inline sha256block_arm64.s]: Look correct?

**Comment by Stebalien** [comment]
bump @harshavardhana

**Comment by Stebalien** [comment]
@harshavardhana is there any chance of this being merged?

**Pull request merged**